src/main/java/witixin/mountables2/entity/Mountable.java

```diff
@@ -369,8 +372,8 @@ else if (this.entityData.get(IS_FLYING) && yComponent < 0 && downSolid)
                 //This getFlyingSpeed = any speed I presume.
                 this.setSpeed(speed);
```

Comment:
The above applies here. Even if this is AI mode (meaning there are no riders at this point) and the movement types might have to cycle in the future, getSpeed() should be the right method to call here.